### PR TITLE
User diet

### PR DESCRIPTION
**Pull request by thawpt**
Addresses Issue #758 
Did:
-Added a new table to the database (dietaryrestriction)
- Added  "Dietary Restriction" to the the modal in My Profile
- Added input field where user is able to type in their dietary restriction as well as update it as needed.

Test:
- Go to My Profile
- Click on Dietary Restriction
- Type in your restrictions, then hit save
- Run test

### CODE REVIEW

File: app/static/js/userProfile.js [hunk 1]
```diff
@@ -158,7 +158,29 @@ $(document).ready(function(){
         }
     });
 
-    setupPhoneNumber("#updatePhone", "#phoneInput")
+  $("#updatePhone").on('click', function() {
+    var username = $(this).data("username")
+    validatePhoneNumber(this, "#phoneInput", username)
+  });
```
Comment (AndersonStettner) [inline app/static/js/userProfile.js]: This is old code that was carried over. It breaks the phone number handler.

File: app/models/dietaryRestriction.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from app.models import *
+from app.models.user import User
+
+class DietaryRestriction(baseModel):
+    user = ForeignKeyField(User)
+    dietRestriction = CharField()
```
Comment (AndersonStettner) [inline app/models/dietaryRestriction.py]: Make this a text field

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/static/js/userProfile.js [hunk 3]
```diff
@@ -158,7 +158,29 @@ $(document).ready(function(){
...
+    let data = {
+      dietInfo: $("#diet").val(),
+      user: $(this).data("user")
+    }
+    $.ajax({
+      type: "POST",
+      url: "/updateDietInformation",
+      data: data,
+      success: function(s){
+        location.reload();
```
Comment (AndersonStettner) [inline app/static/js/userProfile.js]: use reloadWithAccordion() instead of location.reload()

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
The accordion focusing is messed up. With accordion=dietartyInformation the "Dietary Restrictions" accordion is focused and closed while the "Upcoming Events" accordion is focused and open. The "Upcoming Events" accordion should be closed and not focused and the "Dietary Restrictions" accordion should be focused and open. 
image

File: app/models/dietaryRestriction.py (hunk 2 above)
Comment (sreynit02) [reply app/models/dietaryRestriction.py]: fixed

File: app/static/js/userProfile.js (hunk 3 above)
Comment (sreynit02) [reply app/static/js/userProfile.js]: fixed

File: app/static/js/userProfile.js (hunk 1 above)
Comment (sreynit02) [reply app/static/js/userProfile.js]: deleted,

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
Dietary restrictions does not need its won model, we can just add it as a field on the user since it is a one to one mapping.

**Comment by sreynit02** [comment]
> Dietary restrictions does not need its won model, we can just add it as a field on the user since it is a one to one mapping.

Deleted UserDiet table and added a field to User table for diet information

**Comment by AndersonStettner** [comment]
Any further work of this issue/pull request should be done on PR #819

**Comment by sreynit02** [comment]
> Any further work of this issue/pull request should be done on PR #819

Nope

**Review by solijonovam — CHANGES_REQUESTED** [review]
test_event_list.py  is failing.

**Comment by BrianRamsay** [comment]
Merged with mega-branch

**Pull request closed without merging**